sessionctx/stmtctx/stmtctx.go

```diff
@@ -75,6 +75,9 @@ type StatementContext struct {
 	// then cast the float string to int string. Otherwise, we cast string to integer
 	// prefix in a strict way, only extract 0-9 and (+ or - in first bit).
 	CastStrToIntStrict bool
+	// CastInInsert is used to indicate whether the cast is caused by inserting,
+	// using cast() and inserting a value into a table may behave differently.
+	CastInInsert bool
```

Comment:
Can we use `InInsertStmt` to check whether it's caused by `Insert`?

Reply:
No, consider `insert into t values(cast(0xfffffffffffffff as signed))`.
Use InInsertStmt will make `cast(0xfffffffffffffff as signed)` return int64max

Reply:
I tested this case in my MAC.
I changed `if sc.CastInInsert` to `if sc.InInsertStmt` in `datum.go` and it has the same behavior as using `if sc.CastInInsert`.

Reply:
@qw4990 I'm sorry that there is a mistake in my example. 
Try this `insert into t values(cast(0xfffffffffffffffffffff as signed));`

Reply:
I tested it again... they have the same behavior.